### ISSUE TITLE
Federated Identity Credentials fails for User Assigned Identity

### ISSUE THREAD

**Issue by sebafo** [Azure/terraform-provider-azapi#209]
```
Terraform v1.3.3
on darwin_arm64
+ provider registry.terraform.io/azure/azapi v1.0.0
+ provider registry.terraform.io/hashicorp/azuread v2.30.0
+ provider registry.terraform.io/hashicorp/azurerm v3.29.0
```

Code-Snipped:
```terraform
resource "azapi_resource" "symbolicname" {
  type = "Microsoft.ManagedIdentity/userAssignedIdentities/federatedIdentityCredentials@2022-01-31-preview"
  name = "TEST"
  parent_id = azurerm_user_assigned_identity.id.id
  body = jsonencode({
    properties = {
      audiences = [
        "api://AzureADTokenExchange"
      ]
      issuer = "ISSUER_NAME"
      subject = "SUBJECT_NAME"
    }
  })
}
```

When running 'terraform plan' it returns an error message as follows:

Error-Message:
```
│ Error: `resource_id` and `type` are not matched, expect `type` to be 
Microsoft.ManagedIdentity/userAssignedIdentities/federatedIdentityCredentials/federatedIdentityCredentials, but got 
Microsoft.ManagedIdentity/userAssignedIdentities/federatedIdentityCredentials
```

The expected type seems to be a bit odd.

**Comment by ms-henglu** [comment]
Hi @sebafo ,

Thank you for taking time to report this issue!

Unfortunately that I couldn't reproduce it, would you try to create a new one with the following config? It works on my side.

```hcl
resource "azurerm_resource_group" "test" {
  name     = "henglu31"
  location = "westus"
}

resource "azurerm_user_assigned_identity" "test" {
  location            = azurerm_resource_group.test.location
  name                = "henglu31"
  resource_group_name = azurerm_resource_group.test.name
}

resource "azapi_resource" "test" {
  type      = "Microsoft.ManagedIdentity/userAssignedIdentities/federatedIdentityCredentials@2022-01-31-preview"
  name      = "TEST"
  parent_id = azurerm_user_assigned_identity.test.id
  body = jsonencode({
    properties = {
      audiences = [
        "api://AzureADTokenExchange"
      ]
      issuer  = "https://oidc.prod-aks.azure.com/IssuerGUID"
      subject = "system:serviceaccount:ns:svcaccount"
    }
  })
}
```

Indeed the error message is odd, it seems that an incorrect `id` is stored into state, would you please also share the state of the `azapi_resource`?

**Comment by sebafo** [comment]
Hi @ms-henglu,

Thank you very much for your message. You are right, there was something messed up with the terraform state.
The only way to fix it was to edit the terraform state manually.

But what caused this error? I have compared the working from the not working state and I have realized that I accidentally have used a different naming schema in the first place.

Summary of my follow up test regarding the naming of the federated identity credential:

1)	Name: test
Everything works!

2)	Name: something/test (with a ‘/’ in it)
Terraform plan works.
Error when applying: 
```
│ --------------------------------------------------------------------------------
│ RESPONSE 404: 404 Not Found
│ ERROR CODE UNAVAILABLE
│ --------------------------------------------------------------------------------
│ Response contained no body
│ --------------------------------------------------------------------------------
```
=> No federated identity credential created in Azure.

3)	Name: something/somethingtest/test (with two ‘/’ in it)
Terraform plan works.
Error when applying:
```
╷
│ Error: `resource_id` and `type` are not matched, expect `type` to be Microsoft.ManagedIdentity/userAssignedIdentities/federatedIdentityCredentials/test1, but got Microsoft.ManagedIdentity/userAssignedIdentities/federatedIdentityCredentials
│ 
```
=> BUT, Federated identity credential is created with the name: something (everything after first / is omitted)
=> Terraform state broken
The id is stored with the 'full' name like:
`resourceGroups/RESOURCE-GROUP-NAME/providers/Microsoft.ManagedIdentity/userAssignedIdentities/test/federatedIdentityCredentials/something/somethingtest/test`

So the main problem in my case was that the resource was somehow created even though the defined name was not valid.
